### ISSUE TITLE
Run onnxruntime_perf_test with different batch sizes and different precisions with TensorRT execution engine

### ISSUE THREAD

**Issue by mayani-nv** [microsoft/onnxruntime#5358]
**Describe the bug**
Hello experts, I am new to the OnnxRT and was trying to run the `onnxruntime_perf_test` binary for the [RetinaNet model](https://github.com/onnx/models/tree/master/vision/object_detection_segmentation/retinanet) download from Onnx model zoo along with the sample test data which comes along with it. I request if you could please help me with following: 

1. Run the `onnxruntime_perf_test` with TensorRT execution engine for different batch sizes. I guess the default binary assumes the input of batch size of 1. I am using the `-e tensorrt` as mentioned [here](https://github.com/microsoft/onnxruntime/blob/master/docs/execution_providers/TensorRT-ExecutionProvider.md)

2. Run the` onnxruntime_perf_test `with` TensorRT` execution engine with FP16 precision. I used this page and set the environmental variable named` ORT_TENSORRT_FP16_ENABLE` 1 in my Windows environment. Is it enough or does something else needs to be changed somewhere else? 

3. Is there any way to validate the accuracy in results if I switch from one execution engine to other (for eg. using `-e cuda` instead of `-e tensorrt`)

In short, my goal is to evaluate what combination of batch size and precision will provide the best performance for the RetinaNet model? 

**Urgency**
none

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows-Server 2016
- ONNX Runtime installed from (source or binary): Source
- ONNX Runtime version: Master version (1.7) 
- Python version: 3.7
- Visual Studio version (if applicable): VS 2017 Community edition
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 8
- GPU model and memory: Tesla P40

Thanks in advance !

**Comment by jywu-msft** [comment]
it looks like for that particular model, the shape of the input is hardcoded to [1,3,480,640]
thus the model itself only supports batch size 1.
for comparing output between CUDA EP and TensorRT EP, you may want to use onnxruntime python bindings and compare the outputs. onnxruntime_perf_test doesn't save the outputs.

**Comment by mayani-nv** [comment]
@jywu-msft thanks for the reply. I am not sure if I understand it right. I know that the shape of the test input data which comes along with the model is [1,3,480,640] but the github page of the RetinaNet  mentions " The model expects mini-batches of 3-channel input images of shape (N x 3 x H x W), where N is batch size." So, my question was that how do I provide these N input images?

**Comment by jywu-msft** [comment]
> @jywu-msft thanks for the reply. I am not sure if I understand it right. I know that the shape of the test input data which comes along with the model is [1,3,480,640] but the github page of the RetinaNet mentions " The model expects mini-batches of 3-channel input images of shape (N x 3 x H x W), where N is batch size." So, my question was that how do I provide these N input images?

I think the documentation is inaccurate. While the original RetinaNet model may support variable batch size, and H and W.
it looks like the model that was exported/converted and uploaded to the onnx model zoo is only supporting batch size of 1. H of 480 and W of 640
if you examine the input shape of the model (for example using netron)
you'll see the input shape is hardcoded to [1,3,480,640]

normally if the model supports dynamic batch, then you can just feed onnxruntime_perf_test a tensor with batch dimension > 1.
the tool has no notion of "batch size", it just accepts whatever tensor that you feed it. (the tensor itself can represent batches of input data)

**Comment by mayani-nv** [comment]
@jywu-msft Thanks for the explanation and solving my doubts.
